skyportal/handlers/api/internal/source_views.py

```diff
@@ -32,7 +33,24 @@ def get(self):
                      [g.id for g in self.current_user.accessible_groups]))))
              .filter(SourceView.created_at >= cutoff_day)
              .order_by(desc('views')).limit(max_num_sources))
-        return self.success(data=q.all())
+
+        thumbnail_order = ['new', 'ref', 'sub', 'sdss', 'dr8']
+        sources = []
+        for view, obj_id in q.all():
+            s = Source.get_obj_if_owned_by(  # Returns Source.obj
+                obj_id, self.current_user,
+                options=[joinedload(Source.obj)
+                         .joinedload(Obj.thumbnails)])
+
+            public_url = ""
+            for thumbtype in thumbnail_order:
+                for thumbnail in s.thumbnails:
+                    if (thumbnail.type == thumbtype) and (public_url==""):
+                        public_url = thumbnail.public_url
```

Comment:
If you need to sort thumbnails, you can use something like:

```
def tIndex(t): 
    return thumbnail_order.index(t) if t in thumbnail_order else len(thumbnail_order)

for thumbnail in sorted(s.thumbnails, key=lambda t: tIndex(t)):
    ...
```

But, I can't quite figure out what this is supposed to do: grab the first public URL in the order of `new`, `ref` `sub`, etc.?

If so, you could do something like:

```
def first_public_url(thumbnails):
    urls = [t.public_url for t in sorted(s.thumbnails, key=lambda t: tIndex(t))]
    return urls[0] if urls else ""
```

Reply:
Yes much clearer. I pushed this.